File: backend/app/dtos/insurance.py
from pydantic import BaseModel, Field


class InsuranceResponse(BaseModel):
    """DTO for insurance response."""

    id: int
    number: str
    title: str
    description: str | None = None

    class Config:
        from_attributes = True


class InsuranceListResponse(BaseModel):
    """DTO for paginated insurance list response."""

    items: list[InsuranceResponse]
    total: int
    page: int
    page_size: int
    total_pages: int

File: frontend/app/shared/types/insurance.ts
import { z } from "zod";

// Insurance schema
export const insuranceSchema = z.object({
  id: z.number(),
  number: z.string(),
  title: z.string(),
  description: z.string().nullable(),
});

export type Insurance = z.infer<typeof insuranceSchema>;

// Paginated insurance response
export const paginatedInsuranceSchema = z.object({
  items: z.array(insuranceSchema),
  total: z.number(),
  page: z.number(),
  page_size: z.number(),
  total_pages: z.number(),
});

export type PaginatedInsurance = z.infer<typeof paginatedInsuranceSchema>;
